lib/ar2dto/dto.rb

```diff
@@ -4,13 +4,30 @@ module AR2DTO
   class DTO
     class << self
       def [](original_model)
-        Class.new(self) do |_klass|
-          attr_accessor(*original_model.attribute_names)
+        Class.new(self) do
+          attr_reader(*original_model.attribute_names)
+
+          private
+
+          attr_writer(*original_model.attribute_names)
```

Comment:
What was is this change about moving the writer to a private method doing?

Reply:
we don't want the DTOs to be editable so I made the setter private. Do you think it should be public?

Reply:
Then I saw the tests and understood this.

My concern is that some people might want to use DTOs also in the other direction.

E.g. From a certain client, initialize a new DTO that would be transferred to another component/system, in order for that component/system to persist a new record. So for this usage they might want to be able to set attributes too.

Another option could be to make this a global configuration too.